components/camera/align/homography.go

```diff
@@ -71,6 +71,18 @@ func (cfg *homographyConfig) Validate(path string) ([]string, error) {
 	if cfg.Depth == "" {
 		return nil, resource.NewConfigValidationFieldRequiredError(path, "depth_camera_name")
 	}
+
+	if cfg.CameraParameters != nil {
+		if cfg.CameraParameters.Height < 0 || cfg.CameraParameters.Width < 0 {
+			return nil, fmt.Errorf(
+				"align_color_depth_homography needs Width and Height fields set in intrinsic_parameters."+
+					"Got illegal zero or negative dimensions (%d, %d",
+				cfg.CameraParameters.Width,
+				cfg.CameraParameters.Height,
+			)
+		}
+	}
+
```

Comment:
match constructor. though it is weird to have that check there and not in validate.